### PR TITLE
CE-3325 Handle revisions made by Anonymous users

### PR DESCRIPTION
**Pull request by adamkarminski**
@hakubo @rogatty @kamilkoterba

### CODE REVIEW

File: front/main/app/components/user-avatar.js [hunk 1]
```diff
@@ -18,5 +18,7 @@ export default Ember.Component.extend({
 			title: this.get('profileName'),
 		});
 	}),
-	shouldWrapInHref: true
+	shouldWrapInHref: Ember.computed('anonymous', function () {
```
Comment (rogatty) [inline front/main/app/components/user-avatar.js]:
```
shouldWrapInHref: Ember.computed.not('anonymous')
```
http://emberjs.com/api/classes/Ember.computed.html#method_not

File: front/main/app/models/article-diff.js [hunk 2]
```diff
@@ -76,13 +74,26 @@ ArticleDiffModel.reopenClass({
...
+				 * if it is not an anonymous one (userId !== 0). Otherwise, just resolve */
+				if (userId === 0) {
+					model.set('anonymous', true);
+					resolve(model);
+				} else {
+					UserModel.find({userId}).then((user) => {
+						model.setProperties({
+							user: user.name,
+							useravatar: user.avatarPath
+						});
```
Comment (rogatty) [inline front/main/app/models/article-diff.js]: Should we set `anonymous: false`?

File: front/main/app/models/article-diff.js [hunk 3]
```diff
@@ -76,13 +74,26 @@ ArticleDiffModel.reopenClass({
...
 						timestamp: revision.timestamp,
 						title: page.title,
-						user: user.name,
-						useravatar: user.avatarPath
-					}));
-				}).catch((error) => {
-					Ember.Logger.error(error);
-					reject(error);
-				});
+						user: i18n.t('diff.anonymous', {ns: 'recent-wiki-activity'})
```
Comment (rogatty) [inline front/main/app/models/article-diff.js]: It looks strange here. How about moving it inside `if (userId === 0) {`?

File: front/main/app/templates/components/user-avatar.hbs [hunk 4]
```diff
@@ -12,5 +12,9 @@
 	{{/if}}
 </figure>
 {{#if displayProfileName}}
-	<a href="{{profileUrl}}" class="avatar-username">{{profileName}}</a>
+	{{#if shouldWrapInHref}}
+		<a href="{{profileUrl}}" class="avatar-username">{{profileName}}</a>
+	{{else}}
+		{{profileName}}
```
Comment (rogatty) [inline front/main/app/templates/components/user-avatar.hbs]: Is it still "user-avatar" if it's for anonymous and we don't display the avatar? ;)

Comment (adamkarminski) [inline front/main/app/templates/components/user-avatar.hbs]: It is an anonymous user, still a user. :)

**Comment by rogatty** [comment]
LGTM

**Pull request merged**